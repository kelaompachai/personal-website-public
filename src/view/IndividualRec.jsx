import React from 'react';

function IndividualRec(props) {
  const { giver, relation, text } = props;
  return (
    <div className="flex justify-center items-center min-h-72 my-2.5">
      <div className="h-[26rem] flex flex-col max-w-xl bg-well-read-200 rounded-3xl">
        <div className="flex items-center justify-center">
          <div className="cheese flex-grow-0 mx-5 text-gray-800 font-medium">{giver}</div>
          {/* <div className="cheese bg bg-black h-px w-4"></div> */}
          <div className="cheese flex-grow-0 mx-5 text-gray-800 font-light">{relation}</div>
        </div>
        <div className="flex items-center justify-center flex-grow">
          <p className="cheese p-5 text-center text-gray-600 font-light">{text}</p>
        </div>
      </div>
    </div>
  );
}

// text-gray-600

export default IndividualRec;
