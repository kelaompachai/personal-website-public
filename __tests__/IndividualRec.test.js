import { screen, render } from '@testing-library/react';
import IndividualRec from '../src/view/IndividualRec';

describe('IndividualRec renders properly', () => {
  it('looks good', () => {
    expect(render(<IndividualRec giver="Chesterton" relation="brother" text="I am a huge fan of Joe." />).baseElement).toMatchSnapshot();

    // make sure props are in the document as text
    expect(screen.queryByText(/Chesterton/)).toBeInTheDocument();
    expect(screen.queryByText(/brother/)).toBeInTheDocument();
    expect(screen.queryByText(/I am a huge fan of Joe/)).toBeInTheDocument();
  });
});
