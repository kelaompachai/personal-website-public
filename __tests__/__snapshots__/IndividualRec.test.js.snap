// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndividualRec renders properly looks good 1`] = `
<body>
  <div>
    <div
      class="flex justify-center items-center min-h-72 my-2.5"
    >
      <div
        class="h-[26rem] flex flex-col max-w-xl bg-well-read-200 rounded-3xl"
      >
        <div
          class="flex items-center justify-center"
        >
          <div
            class="cheese flex-grow-0 mx-5 text-gray-800 font-medium"
          >
            Chesterton
          </div>
          <div
            class="cheese flex-grow-0 mx-5 text-gray-800 font-light"
          >
            brother
          </div>
        </div>
        <div
          class="flex items-center justify-center flex-grow"
        >
          <p
            class="cheese p-5 text-center text-gray-600 font-light"
          >
            I am a huge fan of Joe.
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
